server/common/transport-shared/src/main/java/org/kaaproject/kaa/server/transport/AbstractKaaTransport.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2014 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaaproject.kaa.server.transport;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.text.MessageFormat;
+import java.util.Arrays;
+
+import org.apache.avro.specific.SpecificRecordBase;
+import org.kaaproject.kaa.common.avro.AvroByteArrayConverter;
+import org.kaaproject.kaa.server.transport.message.MessageHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract implementation of {@link Transport} that handles deserialization of
+ * binary configuration.
+ * 
+ * @author Andrew Shvayka
+ *
+ * @param <T>
+ *            specific configuration record type
+ */
+public abstract class AbstractKaaTransport<T extends SpecificRecordBase> implements Transport {
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractKaaTransport.class);
+    private static final Charset UTF8 = Charset.forName("UTF-8");
+    protected static final int SIZE_OF_INT = 4;
+
+    /**
+     * Message handler
```

Comment:
This is just a variable comment. You can review javadoc of MessageHandler class for more information.